### PR TITLE
Fix doc comment references among other new lints

### PR DESCRIPTION
**Pull request by kevmoo**
Bump min SDK to 2.18

### CODE REVIEW

File: lib/src/algorithms.dart [hunk 1]
```diff
@@ -231,12 +233,12 @@ void mergeSort<E>(List<E> elements,
   var secondLength = end - middle;
   // secondLength is always the same as firstLength, or one greater.
   var scratchSpace = List<E>.filled(secondLength, elements[start]);
-  E Function(E) id = identity;
```
Comment (kevmoo) [inline lib/src/algorithms.dart:234]: This is a fun one!

File: lib/src/combined_wrappers/combined_iterable.dart [hunk 2]
```diff
@@ -17,7 +17,7 @@ class CombinedIterableView<T> extends IterableBase<T> {
   /// The iterables that this combines.
   final Iterable<Iterable<T>> _iterables;
 
-  /// Creates a combined view of [iterables].
+  /// Creates a combined view of [_iterables].
```
Comment (kevmoo) [inline lib/src/combined_wrappers/combined_iterable.dart:20]: type is not public anyway, but the lint flags this reference, for obvious reasons

File: lib/src/algorithms.dart [hunk 3]
```diff
@@ -76,11 +76,13 @@ int lowerBound<E>(List<E> sortedList, E value, {int Function(E, E)? compare}) {
...
-/// Elements are compared using the [compare] function of the [keyOf] property of
-/// the elements.
-/// If the list isn't sorted according to this order, the result is unpredictable.
+/// Elements are compared using the [compare] function of the [keyOf] property
+/// of the elements.
+/// If the list isn't sorted according to this order, the result is
+/// unpredictable.
 ///
-/// Returns [sortedList.length] if all the items in [sortedList] are before [value].
+/// Returns [sortedList]`.length` if all the items in [sortedList] are before
```
Comment (natebosch) [inline lib/src/algorithms.dart:84]:
```suggestion
/// Returns the length of [sortedList] if all the items in [sortedList] are before
```

File: lib/src/iterable_extensions.dart [hunk 4]
```diff
@@ -568,10 +570,10 @@ extension IterableExtension<T> on Iterable<T> {
     return true;
   }
 
-  /// Contiguous [slice]s of [this] with the given [length].
+  /// Contiguous returns slices of `this` with the given [length].
```
Comment (natebosch) [inline lib/src/iterable_extensions.dart:573]:
```suggestion
  /// Contiguous slices of `this` with the given [length].
```

File: test/functions_test.dart [hunk 5]
```diff
@@ -317,47 +318,49 @@ void main() {
...
-          }),
-          equals([
-            // This is expected to return *a* topological ordering, but this isn't
-            // the only valid one. If the function implementation changes in the
-            // future, this test may need to be updated.
-            {'a', 'b', 'e'},
-            {'c', 'd', 'h'},
-            {'f', 'g'},
-          ]));
+        stronglyConnectedComponents({
```
Comment (natebosch) [inline test/functions_test.dart:321]: Are we missing some config to check formatting on PRs? I'm surprised this got through.

Comment (kevmoo) [reply test/functions_test.dart:321]: I added a trailing comma to get things below 80 cols.

**Comment by kevmoo** [comment]
PTAL @natebosch

File: lib/src/algorithms.dart (hunk 3 above)
Comment (natebosch) [reply lib/src/algorithms.dart]: This wasn't changed?

Comment (natebosch) [reply lib/src/algorithms.dart]: oh, nvmd, the pattern shows up in two places.

File: lib/src/algorithms.dart [hunk 6]
```diff
@@ -65,7 +65,7 @@ int binarySearchBy<E, K>(List<E> sortedList, K Function(E element) keyOf,
 /// If [compare] is omitted, this defaults to calling [Comparable.compareTo] on
 /// the objects. In this case, the objects must be [Comparable].
 ///
-/// Returns [sortedList.length] if all the items in [sortedList] compare less
+/// Returns [sortedList]`.length` if all the items in [sortedList] compare less
```
Comment (natebosch) [inline lib/src/algorithms.dart:68]:
```suggestion
/// Returns the length of [sortedList] if all the items in [sortedList] compare less
```

**Comment by kevmoo** [comment]
@natebosch – PTAL 😄

File: test/wrapper_test.dart [hunk 7]
```diff
@@ -169,7 +164,7 @@ bool boolFunc(dynamic x) => true;
 Iterable<dynamic> expandFunc(dynamic x) => [x];
 dynamic foldFunc(dynamic previous, dynamic next) => previous;
 int compareFunc(dynamic x, dynamic y) => 0;
-var val = 10;
+int val = 10;
```
Comment (natebosch) [inline test/wrapper_test.dart:167]: why was this change necessary?

**Review by natebosch — APPROVED** [review]

File: test/wrapper_test.dart (hunk 7 above)
Comment (natebosch) [reply test/wrapper_test.dart:167]: Ah. I don't think we should be turning that one on.
